iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/IcebergTableUtil.java

```diff
@@ -106,4 +111,61 @@ public static PartitionSpec spec(Configuration configuration, Schema schema) {
     });
     return builder.build();
   }
+
+  public static void updateSpec(Configuration configuration, Table table) {
+    // get the new partition transform spec
+    PartitionSpec newPartitionSpec = spec(configuration, table.schema());
+    if (newPartitionSpec == null) {
+      LOG.debug("Iceberg Partition spec is not updated due to empty partition spec definition.");
+      return;
+    }
+
+    List<String> newPartitionNames =
+        newPartitionSpec.fields().stream().map(PartitionField::name).collect(Collectors.toList());
+    List<String> currentPartitionNames = table.spec().fields().stream().map(PartitionField::name)
+        .collect(Collectors.toList());
+    List<String> intersectingPartitionNames =
+        currentPartitionNames.stream().filter(newPartitionNames::contains).collect(Collectors.toList());
+
+    // delete those partitions which are not present among the new partion spec
+    UpdatePartitionSpec updatePartitionSpec = table.updateSpec();
+    currentPartitionNames.stream().filter(p -> !intersectingPartitionNames.contains(p))
+        .forEach(updatePartitionSpec::removeField);
+    updatePartitionSpec.apply();
+
+    // add new partitions which are not yet present
+    List<PartitionTransform.PartitionTransformSpec> partitionTransformSpecList = SessionStateUtil
+        .getResource(configuration, hive_metastoreConstants.PARTITION_TRANSFORM_SPEC)
+        .map(o -> (List<PartitionTransform.PartitionTransformSpec>) o).orElseGet(() -> null);
+    IntStream.range(0, partitionTransformSpecList.size())
+        .filter(i -> !intersectingPartitionNames.contains(newPartitionSpec.fields().get(i).name()))
+        .forEach(i -> {
+          PartitionTransform.PartitionTransformSpec spec = partitionTransformSpecList.get(i);
+          switch (spec.transformType) {
```

Comment:
This `switch` is very similar to the one where we are converting the json to the spec. Do we have a way to reuse code?

Reply:
I'm not sure we can combine the two switch statements.

Reply:
PartitionSpecVisitor would be the class which we should implement in this case (to be honest in both cases)